app/apis/api_application.py

```diff
@@ -58,6 +63,14 @@ class ApiApplications(Resource):
     @app_info_namespace.marshal_list_with(app_info)
     def get(self):
         """get_applications"""
+        if Auth.enabled():
+            user_id = get_jwt_identity()
+            uobj = db.session.query(User).filter(User.user_id == user_id).one()
+            # applications which don't have users are also accessible
+            application_ids = db.session.query(ApplicationUserRole.application_id).distinct().all()
+            ids = [application_id for application_id, in application_ids]
+            applications = db.session.query(Application).filter(~Application.application_id.in_(ids)).all()
+            return [assoc.application for assoc in uobj.applications] + applications
         return Application.query.all()
```

Comment:
Are these return values same format?
`[assoc.application for assoc in uobj.applications] + applications` and `Application.query.all()`.

Reply:
Yes, both values are the `list` of `Application`.